runner-core/src/main/java/org/apache/apisix/plugin/runner/handler/HTTPReqCallHandler.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.apisix.plugin.runner.handler;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Queue;
+import java.util.Set;
+
+import com.google.common.cache.Cache;
+import io.github.api7.A6.Err.Code;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.CollectionUtils;
+import lombok.RequiredArgsConstructor;
+
+import org.apache.apisix.plugin.runner.A6Conf;
+import org.apache.apisix.plugin.runner.A6ErrResponse;
+import org.apache.apisix.plugin.runner.A6Request;
+import org.apache.apisix.plugin.runner.ExtraInfoRequest;
+import org.apache.apisix.plugin.runner.ExtraInfoResponse;
+import org.apache.apisix.plugin.runner.HttpRequest;
+import org.apache.apisix.plugin.runner.HttpResponse;
+import org.apache.apisix.plugin.runner.constants.Constants;
+import org.apache.apisix.plugin.runner.filter.PluginFilter;
+import org.apache.apisix.plugin.runner.filter.PluginFilterChain;
+
+@RequiredArgsConstructor
+public class HTTPReqCallHandler extends SimpleChannelInboundHandler<A6Request> {
```

Comment:
It seems that the object corresponding to this class is a singleton, so the `queue`, `currReq`, `currResp` and other variables bound to a certain HTTP request will have problems in case of concurrent requests

Reply:
It is necessary to use a request ID (e.g. `confToken`) to distinguish these variables that are bound to a certain http request process, and to clean up these context variables  an after the process is completed.

Reply:
Based on the description here, it seems I don't need to control concurrent security?https://github.com/netty/netty/wiki/New-and-noteworthy-in-4.0#well-defined-thread-model

The handler is not @Sharable, I thought it was on the same channel, not two requests come in at the same time, but the next request is processed after the current one. So I cleaned up the queue, currReq, currResp when the HTTPReqCall came in.

Reply:
I don't know if I've got it wrong.